### PR TITLE
Alternative keyboard shortcuts for macOS

### PR DESCRIPTION
**Pull request by ghost**
Follow up of https://github.com/godotengine/godot/pull/14892.

Adds several built-in alternative shortcuts for Mac (very useful on MacBooks).

They're added additionally and do not replace Godot's default shortcuts (which means a Windows Godotian on macOS could still use the Windows shortcuts, keyboard provided):

bildschirmfoto 2017-12-24 um 15 24 14

If a user changes a shortcut to non-default, the alternative shortcut will also go away (the latter will thus be also free to be set to another user-defined action).

The custom handling happens purely inside `ShortCut`, the rest of Godot should not see this replacement handling happening.

Hard coded alternative shortcuts for Mac:

Play: &#x2318; B
Play Scene: &#x2318; R
Play Custom Scene: &#x2318; &#x21E7; R
Pause Scene:  ^ &#x2318; Y
Stop: &#x2318; .

Find Next: &#x2318; G
Find Previous: &#x2318; &#x21E7; G

### CODE REVIEW

File: scene/gui/shortcut.cpp [hunk 1]
```diff
@@ -44,13 +91,29 @@ Ref<InputEvent> ShortCut::get_shortcut() const {
...
+	if (shortcut.is_valid()) {
+		String text = shortcut->as_text();
+
+		Ref<InputEvent> alt = _get_alternative(shortcut);
+		if (alt.is_valid())
+			text += " / " + alt->as_text();
+
+		return text;
+	}
 	else
```
Comment (volzhs) [inline scene/gui/shortcut.cpp]: clang format error here. `else` should be at upper line (116)

File: scene/gui/shortcut.h [hunk 2]
```diff
@@ -39,6 +39,8 @@ class ShortCut : public Resource {
 
 	Ref<InputEvent> shortcut;
 
+    Ref<InputEvent> _get_alternative(const Ref<InputEvent> &p_shortcut) const;
```
Comment (volzhs) [inline scene/gui/shortcut.h]: clang format error here. should use tab instead of spaces.

**Comment by icnikerazah** [comment]
please these shortcuts for macbooks
delete: backspace
delete with dialog: shift  + backspace

**Comment by reduz** [comment]
I would replace them, not add alternatives

**Comment by poke1024** [comment]
All basic macOS shortcuts are in place now; no longer as alternatives but as replacements.

@icnikerazah I've added Command + Backspace as a general replacement for Delete; I'm not sure about the dialog part - having a dialog or not for some deletion operations should probably be a different PR, as it needs to be changed in different parts of the code defining different deletion shortcuts.

**Comment by icnikerazah** [comment]
@poke1024 thanks

File: editor/editor_settings.cpp [hunk 3]
```diff
@@ -1316,8 +1316,44 @@ Ref<ShortCut> ED_GET_SHORTCUT(const String &p_path) {
...
+struct ShortCutMapping {
+	const char *path;
+	uint32_t keycode;
+};
+
 Ref<ShortCut> ED_SHORTCUT(const String &p_path, const String &p_name, uint32_t p_keycode) {
 
+#ifdef OSX_ENABLED
+	static const ShortCutMapping macos_mappings[] = {
+		{ "editor/play", KEY_MASK_CMD | KEY_B },
```
Comment (icnikerazah) [inline editor/editor_settings.cpp]: Both have the same key binding(command+b). the line is cloned when the project is run.
https://github.com/godotengine/godot/blob/109242c011255757c981f634a33f59268a113f67/editor/plugins/script_text_editor.cpp#L1648

**Pull request merged**